granulate_utils/config_feeder/client/http_client.py

```diff
@@ -0,0 +1,47 @@
+from typing import Any, Dict, Optional, cast
+
+from pydantic import BaseModel
+from requests import Session
+from requests.exceptions import ConnectionError, JSONDecodeError
+
+from granulate_utils.config_feeder.client.exceptions import APIError, ClientError
+from granulate_utils.config_feeder.core.errors import raise_for_code
+
+DEFAULT_API_SERVER_ADDRESS = "https://api.granulate.io/config-feeder/api/v1"
+DEFAULT_REQUEST_TIMEOUT = 3
+
+
+class HttpClient:
+    def __init__(self, token: str, server_address: Optional[str]) -> None:
```

Comment:
replace `token` with `headers: Dict[str, str]` here, this will allow collectors to pass custom headers when needed.

Reply:
I actually like it when it's explicit, as it (token) is a mandatory param (better business-wise reflection). We can add additional custom params if needed later.

Reply:
`HttpClient` is not longer passed as a param to the collectors. If it's needed, I'll consider this.